core/src/main/java/com/radixdlt/p2p/hostip/NetworkQueryHostIp.java

```diff
@@ -148,33 +148,39 @@ public VotedResult queryNetworkHosts() {
   }
 
   VotedResult get() {
-    return publicIp((count() + 1) / 2); // Round up
-  }
-
-  VotedResult publicIp(int threshold) {
     // Make sure we don't DoS the first one on the list
     Collections.shuffle(this.hosts);
     log.debug("Using hosts {}", this.hosts);
     final Map<HostIp, AtomicInteger> successCounts = Maps.newHashMap();
     final ImmutableMap.Builder<URL, Result<HostIp, IOException>> queryResults =
         ImmutableMap.builder();
+    int maxCount = 0;
+    Optional<HostIp> maxResult = Optional.empty();
     for (URL url : this.hosts) {
       final Result<HostIp, IOException> result = query(url);
       queryResults.put(url, result);
-      final Optional<HostIp> success = result.toOptional();
-      if (success.isPresent()) {
+      if (result.isSuccess()) {
         int newCount =
             successCounts
-                .computeIfAbsent(success.get(), k -> new AtomicInteger())
+                .computeIfAbsent(result.unwrap(), k -> new AtomicInteger())
                 .incrementAndGet();
-        if (newCount >= threshold) {
-          log.debug("Found address {}", success.get());
-          return new VotedResult(success, queryResults.build());
+        if (newCount > maxCount) {
+          maxCount = newCount;
+          maxResult = result.toOptional();
         }
       }
     }
-    log.debug("No suitable address found");
-    return new VotedResult(Optional.empty(), queryResults.build());
+    if (successCounts.isEmpty()) {
+      log.warn("No suitable address found");
+    }
+    if (successCounts.size() > 1) {
+      String votes =
+          successCounts.keySet().stream()
+              .map(key -> key + "=" + successCounts.get(key))
+              .collect(Collectors.joining(", ", "{", "}"));
+      log.warn("More than 1 IP address found: " + votes);
```

Comment:
MINOR: This is not super clear to me as a node runner whether it's something to be worried about - could we instead say something like:
```java
log.warn(String.format(
    "Attempts to resolve this node's public IP address with external resolution services resulted in more than one distinct IP address. The node will continue by using the most common: %s. The full list of resolved addresses is: %s",
    maxResult.get(),
    votes,
));
```